src/qibolab/instruments/zhinst/sweep.py

```diff
@@ -0,0 +1,139 @@
+                    and sweeper.parameter is Parameter.amplitude
+                ):
+                    sweep_param = lo.SweepParameter(
+                        values=sweeper.values / max(sweeper.values)
+                    )
+                    channel_sweeps.append(
+                        (
+                            measure_channel_name(qubits[pulse.qubit]),
+                            sweeper.parameter,
+                            sweep_param,
+                        )
+                    )
+                else:
+                    sweep_param = lo.SweepParameter(values=copy(sweeper.values))
+                    pulse_sweeps.append((pulse, sweeper.parameter, sweep_param))
+                parallel_sweeps.append((sweeper, sweep_param))
+
+            for qubit in sweeper.qubits or []:
+                if sweeper.parameter is not Parameter.bias:
+                    raise ValueError(
+                        f"Sweeping {sweeper.parameter.name} for {qubit} is not supported"
+                    )
+                sweep_param = lo.SweepParameter(
+                    values=sweeper.values + qubit.flux.offset
+                )
+                channel_sweeps.append((qubit.flux.name, sweeper.parameter, sweep_param))
+                parallel_sweeps.append((sweeper, sweep_param))
+
+            for coupler in sweeper.couplers or []:
+                if sweeper.parameter is not Parameter.bias:
+                    raise ValueError(
+                        f"Sweeping {sweeper.parameter.name} for {coupler} is not supported"
+                    )
+                sweep_param = lo.SweepParameter(
+                    values=sweeper.values + coupler.flux.offset
+                )
+                channel_sweeps.append(
+                    (coupler.flux.name, sweeper.parameter, sweep_param)
+                )
+                parallel_sweeps.append((sweeper, sweep_param))
+
+        self._pulse_sweeps = pulse_sweeps
+        self._channel_sweeps = channel_sweeps
+        self._parallel_sweeps = parallel_sweeps
+
+    def sweeps_for_pulse(
+        self, pulse: Pulse
+    ) -> list[tuple[Parameter, lo.SweepParameter]]:
+        return [item[1:] for item in self._pulse_sweeps if item[0] == pulse]
+
+    def sweeps_for_channel(self, ch: str) -> list[tuple[Parameter, lo.SweepParameter]]:
+        return [item[1:] for item in self._channel_sweeps if item[0] == ch]
+
+    def sweeps_for_sweeper(self, sweeper: Sweeper) -> list[lo.SweepParameter]:
+        return [item[1] for item in self._parallel_sweeps if item[0] == sweeper]
+
+    def channel_sweeps_for_sweeper(
+        self, sweeper: Sweeper
+    ) -> list[tuple[str, Parameter, lo.SweepParameter]]:
+        return [
+            item
+            for item in self._channel_sweeps
+            if item[2] in self.sweeps_for_sweeper(sweeper)
+        ]
+
+    def channels_with_sweeps(self) -> set[str]:
+        return {ch for ch, _, _ in self._channel_sweeps}
```

Comment:
Not sure if all of them are strictly needed, but one line of docs for each of them wouldn't hurt ;)

Reply:
I am not sure if I can add more than the name and the type annotations already tell :D however I can add a bit more doc to the class itself to explain what is happening.

Reply:
Then keep it as minimal as possible, and maybe add a paragraph to describe them collectively in the class docstring.